web-api/src/personalization/routes.rs

```diff
@@ -461,6 +463,10 @@ impl SemanticSearchQuery {
             Err(BadRequest::from("count has to be at least 1").into())
         }
     }
+
+    fn min_similarity(&self) -> Option<f32> {
+        self.min_similarity.map(|value| value.clamp(0., 1.))
```

Comment:
:wrench: see the other comment
```suggestion
        self.min_similarity.map(|value| value.clamp(-1., 1.))
```